app/scripts/lib/session.js

```diff
@@ -41,7 +41,22 @@ define([
         // ignore the parse error.
       }
 
-      this.set(values);
+      // Update new values, without overwriting items on the prototype.
```

Comment:
On reflection, "sync" suggests a two-way synchronisation between the session and its backing store.  How about "reload"?